spec/engines/domain_scope_options_builder_spec.rb

```diff
@@ -30,5 +30,17 @@
         )
       end
     end
+
+    context 'when a sitelimit is passed' do
```

Comment:
it's a good practice to define the context within the context itself, rather than the expectation. That allows you to re-use the context conditions in subsequent specs. So in this case, you could do:
```ruby
    context 'when a sitelimit is passed' do
      let(:args) do
        { site: affiliate, site_limits: 'https://nps.gov/foo' }
      end
 
      it 'strips out the protocols' do
        expect(DomainScopeOptionsBuilder.
          build(args)).to eq(
...
```
However, we can clean that spec up even more by setting a `subject`:
```ruby
  describe '.build' do
    let(:affiliate) { affiliates(:basic_affiliate) }
    subject(:build) { DomainScopeOptionsBuilder.build(args) }
.
.
.
    context 'when a sitelimit is passed' do
      let(:args) do
        { site: affiliate, site_limits: 'https://nps.gov/foo' }
      end
 
      it 'strips out the protocols' do
        expect(build).to eq(
            included_domains: ['nps.gov'],
            excluded_domains: [],
            scope_ids: [],
            site_limits: 'nps.gov/foo'
          )
      end
    end
```

It would also be a good idea to add a spec for `when multiple sitelimits are passed`. That's covered by the integration-y spec in `i14y_search_spec`, but it should be tested in the class doing the work.